lib/prima_auth0_ex/openid_configuration.ex

```diff
@@ -0,0 +1,33 @@
+defmodule PrimaAuth0Ex.OpenIDConfiguration do
+  @moduledoc false
+  @doc """
+    Module for fetching and parsing the [rfc8414](https://www.rfc-editor.org/rfc/rfc8414) openid metadata endpoint
+
+    This allows auth0_ex to be agnostic of the actual openid server
+  """
+
+  @type t :: %__MODULE__{issuer: String.t(), token_endpoint: String.t(), jwks_uri: String.t()}
+
+  @struct_keys [:issuer, :token_endpoint, :jwks_uri]
+  defstruct @struct_keys
+
+  @doc """
+    Fetches the openid metadata.
+
+    Doesn't implement caching and always makes a http request, avoid calling this in hot code paths
+  """
+  @spec fetch(String.t()) :: __MODULE__.t()
+  def fetch(base_url) do
+    url = metadata_url(base_url)
+    %HTTPoison.Response{status_code: status_code, body: meta_body} = Telepoison.get!(url, accept: "application/json")
+
+    true = status_code in 200..299
```

Comment:
I would handle it in a more graceful way since the error code might be a bit confusing from the user perspective. I think we might custom panic this and the `Telepoison.get!` to give a more relevant information to the user

Reply:
Moreover i think it might be better to call this `def fetch!(...` being that this function might panic. Otherwise you can return `{:ok, __MODULE__.t()} | {:error, atom() | String.t()}` and gracefully handle panics

Reply:
I've changed it to `fetch!`. I think panicking in this situation makes sense